TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/redCloseAuto.java

```diff
@@ -0,0 +1,113 @@
+import org.firstinspires.ftc.teamcode.utility.CubeSide;
+import org.firstinspires.ftc.teamcode.utility.FieldSide;
+
+//Names the autonomous on the control hub dropdown.
+@Autonomous(name = "pixelMarksAuto")
+public class redCloseAuto extends LinearOpMode {
+    @Override
+    //This method is needed for teleOp and auto to run at all.
+    public void runOpMode() {
+        //Runs after the robot has been initialized.
+        //Assign vars here and create instances of needed classes.
+
+        telemetry.setAutoClear(false);
+        telemetry.addData("Status", "initializing");
+        telemetry.update();
+
+
+        //Assign drivebase to a instance.
+        Drivebase driveBase = new Drivebase(hardwareMap, this::opModeIsActive);
+        //Assign the camera to a instance of the object.
+        VisionCamera visionCamera = new VisionCamera(hardwareMap, FieldSide.RedClose);
+
+        //Needed for both types of opModes.
+        waitForStart();
+        //Runs while the opmode is running.
+        //This is needed to make a auto or teleOp run.
+        CubeSide side;
+        while ((side = visionCamera.getCubePrediction()) == null) sleep(20);
+
+        //The CubeSide.Left is only for testing. Once done testing change it back to CubeSide.
+        driveToSpike(driveBase, CubeSide.Left);
+
+        //Runs after the stop button has been hit on the control hub.
+    }
+
+    /**
+     * Turns robot to the correct angle and then drives to the center of the spike tape the white pixel's detected on.
+     * That could be the left, middle, or right pixel depending on what the pipeline sees.
+     */
+    public void driveToSpike(Drivebase driveBase, CubeSide side) {
+        //if side is...
+        switch (side) {
+            //if the left side is seen by the pipeline
+            case Left:
+                driveBase.turnAngle(5.0, 0.5, null);
+                driveBase.driveForward(31.5, 0.5, null);
+            case Right:
+                driveBase.turnAngle(20.0, 0.5, null);
+                driveBase.driveForward(29.625, 0.5, null);
+            case Middle:
+                driveBase.turnAngle(10.0, 0.5, null);
+                driveBase.driveForward(26.875, 0.5, null);
+        }
+    }
+
+    /**
+     * Drive robot to the backboard using a path dependent on where the robot already is and the direction it's already
+     * facing then park facing the back glass of the feild.
+     * @param driveBase
+     * @param side
+     */
+    public void driveToBackBoard(Drivebase driveBase, CubeSide side)
+    {
+        //if side is...
+        switch (side){
+            //if the left side is seen by the pipeline then the robot starts facing the 5 degree offset that it was before so...
+            case Left:
+            driveBase.turnAngle(85, 0.5, null);
+            driveBase.driveForward(57, 0.5, null);
+
+            //if the Right side is seen by the pipeline then the robot starts facing the 20 degree offset it was before so...
```

Comment:
Uncapitalize Right here, don't use ellipses in your comments. Commenting is supposed to fill in those blanks.